tests/ServerTest.php

```diff
@@ -410,6 +409,7 @@ public function testGetAllParams()
         $all_params = $this->server->getAllParams([
             'w' => '100',
             'p' => 'small',
+            'invalid' => '1',
```

Comment:
@ADmad I realize I added some fixes that are out of scope of the issue, but this is the test fixing the cache path issue

I believe it shouldn't accept invalid parameters